File: camerax_android/android/src/main/kotlin/dev/zeekr/camerax_android/ml/barcode/BarcodeScanningImpl.kt
package dev.zeekr.camerax_android.ml.barcode

import com.google.mlkit.vision.barcode.BarcodeScanner
import com.google.mlkit.vision.barcode.BarcodeScannerOptions
import com.google.mlkit.vision.barcode.BarcodeScanning
import dev.zeekr.camerax_android.CameraXApiPigeonProxyApiRegistrar
import dev.zeekr.camerax_android.PigeonApiBarcodeScanningApi

class BarcodeScanningImpl(registrar: CameraXApiPigeonProxyApiRegistrar) : PigeonApiBarcodeScanningApi(registrar) {
    override fun getClient1(): BarcodeScanner {
        return BarcodeScanning.getClient()
    }

    override fun getClient2(options: BarcodeScannerOptions): BarcodeScanner {
        return BarcodeScanning.getClient(options)
    }
}

File: camerax_android/lib/src/ml/barcode/barcode_scanning_impl.dart
import 'package:camerax_android/src/camerax_api.g.dart';
import 'package:camerax_platform_interface/camerax_platform_interface.dart';

import 'barcode_scanner_impl.dart';
import 'barcode_scanner_options_impl.dart';

final class BarcodeScanningImpl extends BarcodeScanning {
  static Future<BarcodeScanner> getClient([
    BarcodeScannerOptions? options,
  ]) async {
    final api = options == null
        ? await BarcodeScanningApi.getClient1()
        : await BarcodeScanningApi.getClient2(options.api);
    return BarcodeScannerImpl.internal(api);
  }

  final BarcodeScanningApi api;

  BarcodeScanningImpl.internal(this.api) : super.impl();
}
